### DISCUSSION TITLE
Plain text password in keycloak

### DISCUSSION THREAD

**Discussion by tsyspri** [keycloak/keycloak#22544] [Q&A]
Is that possible to disable hashing in password policy and store plain text password instead in keycloak? I am using 22.0.1 keycloak version.

**Comment by justin-tay** [comment]
There is no easy way to do so as what you are suggesting is very insecure, though I suspect it is possible if you implement a custom password hash SPI. There's probably a better way of achieving what you are trying to do.

**Reply by tsyspri** [reply]
I understand this is not secure way to deal with it. Still wanted to know are there any client configuration/admin settings which we can change to store plain text format password or only way to achieve this is using a custom password hash SPI?

**Reply by justin-tay** [reply]
No there isn't any such setting. By default it can only use PBKDF2. The settings you can configure for passwords is at `Authentication > Policies > Password policy` where you can configure the `Hashing Algorithm` or `Hashing Iterations`

**Comment by christian-2** [comment]
@tsyspri  may I ask what is your use case? 

It seems to me that even if you introduced a custom SP that did no password hashing you had no means of obtaining the password via the Keycloak REST API, and for reading out the password you would have to go in directly e.g. at the database layer instead. Last time I checked, [`GET /{realm}/users/{id}/credentials`](https://www.keycloak.org/docs-api/18.0/rest-api/#_users_resource) returned only `hashIterations`, `algorithm`, and `additionalParameters` in `credentialData`, not the hash itself.

My guess is that your use case involves reading out the password, for if it would remain only in Keycloak's internal use you would not have to care about its hashing, right? Am I mistaken?
